confidant/routes/services.py

```diff
@@ -285,12 +285,19 @@ def map_service_credentials(id):
         except keymanager.ServiceCreateGrantError:
             msg = 'Failed to add grants for {0}.'.format(id)
             logging.error(msg)
+    credentials = credentialmanager.get_credentials(data.get('credentials'))
```

Comment:
also a nit - might be worth making this into a function or adding a comment.  Otherwise it's not obvious what's going on behind the scenes

```
def filter_archived_credentials(credentials):
    filtered_credentials = credentialmanager.get_credentials(data.get('credentials')). # BatchGetItem will automatically remove credentials that don't exist
    return [f.id for f in filtered_credentials]
```

Reply:
We need to use the credentials again later in the function (for the return). I just moved the fetching of them up so that I could also get a list of filtered IDs (that's documented).